S2VX.Game/Story/Note/HoldNote.cs

```diff
@@ -22,20 +23,64 @@ public abstract class HoldNote : S2VXNote {
         [Resolved]
         private S2VXStory Story { get; set; }
 
-        [BackgroundDependencyLoader]
-        private void Load() => AddInternal(SliderPath);
-
-        public Path SliderPath { get; set; } = new() {
+        public SliderPath SliderPath { get; set; } = new() {
             // Show slider path behind note
             Depth = 100,
             Anchor = Anchor.Centre
         };
 
+        [BackgroundDependencyLoader]
+        private void Load() => AddInternal(SliderPath);
+
         protected override void UpdatePosition() {
             UpdateIndicator();
             UpdateSliderPath();
         }
 
+        // These are static helpers for calculating hold related values
+        // TODO: There's probably a better way to share this between HoldApproach and HoldNote for calculating these values
+        // TODO: Optimize these so that they are called once on initialization instead of ran every update loop
+        private static IEnumerable<Vector2> CombineAllCoordinates(Vector2 coordinates, List<Vector2> midCoordinates, Vector2 endCoordinates)
+            => new List<Vector2> { coordinates }.Concat(midCoordinates.Append(endCoordinates));
```

Comment:
The return type of `Append()` is `IEnumerable`. When working with LINQ functions, it can be easier to treat everything as `IEnumerable` since LINQ expects to work with `IEnumerable` as inputs and outputs. Alternatively, you could call `ToList()` at the end and convert it to a list:

```csharp
private static List<Vector2> CombineAllCoordinates(Vector2 coordinates, List<Vector2> midCoordinates, Vector2 endCoordinates)
    => new List<Vector2> { coordinates }.Concat(midCoordinates.Append(endCoordinates)).ToList();
```

But I don't really like this since it feels like it's wasting a step to do nothing.